service/DevDocs/SendAsyncTimeout.md

```diff
@@ -0,0 +1,36 @@
+# SendAsyncTimeout Requirements
+
+## Overview
+
+DeviceClient class is used by the device client callback delegate to prepare the callback results.
+
+## References
+
+##Exposed API
+
+```csharp
+
+sealed class AmqpServiceClient : ServiceClient
+{
+    ...
+    public async override Task SendAsync(string deviceId, Message message, TimeSpan timeout)
```

Comment:
Should we pass a `CancellationToken` instead of a `TimeSpan` here?

Reply:
The AMQP send message we call already takes a Timespan for timeout. Adding the cancellation token may make it harder than need be.

Reply:
I noticed that the `ServiceClient` object is using `CancellationToken` but not TimeSpan so it would be nice to be consistent.
That said, I do see your point: the AMQP library offers the timeout parameter and no cancellationToken. Unless there's a way to cancel that operation whenever the token is cancelled, we could use the TimeSpan in this case.